tools/convert_datasets/a2d2.py

```diff
@@ -0,0 +1,388 @@
+                               label_suffix='_labelTrainIds.png'):
+    '''Creates a new directory structure and symlink existing files into it.
+
+    Required to make the A2D2 dataset conform to the mmsegmentation frameworks
+    expected dataset structure.
+
+    my_dataset
+    └── img_dir
+    │   ├── train
+    │   │   ├── xxx{img_suffix}
+    |   |   ...
+    │   ├── val
+    │   │   ├── yyy{img_suffix}
+    │   │   ...
+    │   ...
+    └── ann_dir
+        ├── train
+        │   ├── xxx{seg_map_suffix}
+        |   ...
+        ├── val
+        |   ├── yyy{seg_map_suffix}
+        ... ...
+
+    Args:
+        a2d2_path: Absolute path to the A2D2 'camera_lidar_semantic' directory.
+        val_ratio: Float value representing ratio of validation samples.
+        test_ratio:
+        label_suffix: Label filename ending string.
+    '''
+
+    # Create new directory structure (if not already exist)
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'train'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'val'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'img_dir', 'test'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'train'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'val'))
+    mmcv.mkdir_or_exist(osp.join(a2d2_path, 'ann_dir', 'test'))
+
+    # Lists containing all images and labels to symlinked
+    img_filepaths = sorted(glob.glob(osp.join(a2d2_path,
+                           '*/camera/cam_front_center/*.png')))
+    ann_filepaths = sorted(glob.glob(osp.join(a2d2_path,
+                           '*/label/cam_front_center/*{}'.format(label_suffix)
+                                              )))
+
+    # Split data according to given ratios
+    total_samples = len(img_filepaths)
+    train_ratio = 1.0 - val_ratio - test_ratio
+
+    train_idx_end = int(np.floor(train_ratio*(total_samples-1)))
+    val_idx_end = train_idx_end + int(np.ceil(val_ratio*total_samples))
+
+    # Create symlinks file-by-file
+    for sample_idx in range(total_samples):
+
+        img_filepath = img_filepaths[sample_idx]
+        ann_filepath = ann_filepaths[sample_idx]
+
+        # Partions string: [generic/path/to/file] [/] [filename]
+        img_filename = img_filepath.rpartition('/')[2]
+        ann_filename = ann_filepath.rpartition('/')[2]
+
+        if sample_idx <= train_idx_end:
+            split = 'train'
+        elif sample_idx <= val_idx_end:
+            split = 'val'
+        else:
+            split = 'test'
+
+        img_symlink_path = osp.join(a2d2_path, 'img_dir', split, img_filename)
+        ann_symlink_path = osp.join(a2d2_path, 'ann_dir', split, ann_filename)
+
+        # NOTE: Can only create new symlinks if no priors ones exists
+        try:
+            symlink(img_filepath, img_symlink_path)
+        except FileExistsError:
+            pass
```

Comment:
In what case the file would exist already?

Reply:
@xvjiarui I imagine handling a scenario where a user might want to repair the converted dataset by re-running the conversion function without getting an error regarding existing symlink files (i.e. do not have to remember to manually delete the symlink directory structure)

Does this potential elimination of error messaging make sense to you?

Of course we can remove it and throw an error instead, reminding the user to delete the old symlink files?